### PR TITLE
[2401, 2431]show popup if uploading faulty file, auth token added in combine docs…

### PR DESCRIPTION
**Pull request by ashishbeehyv**
https://github.com/pucardotorg/dristi/issues/2401
https://github.com/pucardotorg/dristi/issues/2431

## Summary by CodeRabbit

- **New Features**
	- Introduced a `DocumentUploadError` class for improved error handling during document uploads.
	- Enhanced document upload functionality with specific error messages based on document type.

- **Bug Fixes**
	- Improved error handling for document format issues and upload failures, providing clearer user feedback.

- **Refactor**
	- Updated multiple functions to incorporate document type handling, enhancing overall document management and validation.

### CODE REVIEW

**Review by anirudh-0 — APPROVED** [review]

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/errorUtil.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export class DocumentUploadError extends Error {
+  constructor(message, documentType) {
+    super(message);
+    this.documentType = documentType;
+    this.name = "DocumentUploadError";
+  }
+}
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/errorUtil.js:7]: not needed

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/index.js [hunk 2]
```diff
@@ -139,10 +140,34 @@ export const extractFeeMedium = (feeName) => {
...
+  proofOfAcknowledgmentFileUpload: "case.demandnotice.serviceproof",
+  proofOfReplyFileUpload: "case.replynotice",
+  debtLiabilityFileUpload: "case.liabilityproof",
+  condonationFileUpload: "CONDONATION_DOC",
+  swornStatement: "case.affidavit.225bnss",
+  SelectUploadDocWithName: "case.docs",
+  vakalatnamaFileUpload: "VAKALATNAMA_DOC",
+};
+
+const token = localStorage.getItem("token");
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/index.js:162]: we will get token inside function call.

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/index.js [hunk 3]
```diff
@@ -178,17 +203,17 @@
...
-      responseType: "blob", // To handle the response as a Blob
+      responseType: "blob",
     });
     const file = new File([response.data], finalFileName, { type: response.data.type });
     return [file];
   } catch (error) {
     console.error("Error:", error);
-    throw error;
+    throw new DocumentUploadError(`Document upload failed: ${error.message}`, documentsTypeMapping[key]);
   }
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/index.js:218]: not needed

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/index.js [hunk 4]
```diff
@@ -139,10 +140,34 @@
...
+  legalDemandNoticeFileUpload: "case.demandnotice",
+  proofOfDispatchFileUpload: "case.demandnotice.proof",
+  proofOfAcknowledgmentFileUpload: "case.demandnotice.serviceproof",
+  proofOfReplyFileUpload: "case.replynotice",
+  debtLiabilityFileUpload: "case.liabilityproof",
+  condonationFileUpload: "CONDONATION_DOC",
+  swornStatement: "case.affidavit.225bnss",
+  SelectUploadDocWithName: "case.docs",
+  vakalatnamaFileUpload: "VAKALATNAMA_DOC",
+};
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/index.js:160]: this is the requirement

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/index.js [hunk 5]
```diff
@@ -178,17 +203,17 @@
 
   try {
     // ${Urls.CombineDocuments} // check- Should use this but it is causing circular dependency, need to relocate Urls
-    const combineDocumentsUrl = `${window.location.origin}/egov-pdf/dristi-pdf/combine-documents`;
+    const combineDocumentsUrl = `${window.location.origin}/egov-pdf/dristi-pdf/combine-documents?tenantId=${tenantId}`;
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/index.js:206]: not needed

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/citizen/FileCase/EFilingCases.js [hunk 6]
```diff
@@ -1763,10 +1763,13 @@ function EFilingCases({ path }) {
...
       })
-      .catch((error) => {
-        if (extractCodeFromErrorMsg(error) === 413) {
+      .catch(async (error) => {
+        if (error instanceof DocumentUploadError) {
+          toast.error(`${t("DOCUMENT_FORMAT_DOES_NOT_MATCH")} : ${error?.documentType}`);
+        } else if (extractCodeFromErrorMsg(error) === 413) {
           toast.error(t("FAILED_TO_UPLOAD_FILE"));
         } else {
+          console.error("Error:", error);
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/citizen/FileCase/EFilingCases.js:1772]: not needed

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/citizen/FileCase/EfilingValidationUtils.js [hunk 7]
```diff
@@ -1029,10 +1030,14 @@
...
+const onDocumentUpload = async (documentType = "Document", fileData, filename, tenantId) => {
   if (fileData?.fileStore) return fileData;
-  const fileUploadRes = await window?.Digit.UploadServices.Filestorage("DRISTI", fileData, tenantId);
-  return { file: fileUploadRes?.data, fileType: fileData.type, filename };
+  try {
+    const fileUploadRes = await window?.Digit.UploadServices.Filestorage("DRISTI", fileData, tenantId);
+    return { file: fileUploadRes?.data, fileType: fileData.type, filename };
+  } catch (error) {
+    throw new DocumentUploadError(`Document upload failed: ${error.message}`, documentType);
+  }
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/citizen/FileCase/EfilingValidationUtils.js:1040]: this is correct

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/citizen/FileCase/EFilingCases.js [hunk 8]
```diff
@@ -1837,7 +1840,15 @@
...
-      .catch(() => {
+      .catch(async (error) => {
+        if (error instanceof DocumentUploadError) {
+          toast.error(`${t("DOCUMENT_FORMAT_DOES_NOT_MATCH")} : ${error?.documentType}`);
+        } else if (extractCodeFromErrorMsg(error) === 413) {
+          toast.error(t("FAILED_TO_UPLOAD_FILE"));
+        } else {
+          console.error("Error:", error);
+          toast.error(t("SOMETHING_WENT_WRONG"));
+        }
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/citizen/FileCase/EFilingCases.js:1851]: can be done later when we will handle generate pdf api giving error on faulty file

**Pull request merged**